File: leetcode_hot100/94 二叉树的中序遍历/code.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        arr = []

        def Inorder(root: Optional[TreeNode]):
            if (root == None):
                return

            Inorder(root.left)
            arr.append(root.val)
            Inorder(root.right)

        Inorder(root)
        return arr
        

File: leetcode_hot100/二叉树/199 二叉树右视图/code.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function rightSideView(root: TreeNode | null): number[] {

    const arr: number[] = [];
    let depth: number = 0;

    function RightView(root: TreeNode | null, s_depth: number) {
        if (root === null)
            return;
        if (s_depth >= depth) {
            arr.push(root.val);
            depth++;
        }
        RightView(root.right, s_depth+1);
        RightView(root.left, s_depth+1)
    }

    RightView(root,0);
    return arr;
};